app/models/manageiq/providers/ibm_power_hmc/infra_manager/vm.rb

```diff
@@ -65,4 +65,17 @@ def poweroff(_params = {})
   def make_template(_clone_options)
     raise StandardError, "Must be implemented in a subclass"
   end
+
+  def capture_metrics(counters, start_time = nil, end_time = nil)
+    samples = collect_samples(start_time, end_time)
+    process_samples(counters, samples)
+  end
```

Comment:
Wonder if some of this duplication can be moved to a `MetricsCaptureMixin` with this and e.g. `SAMPLE_DURATION` and maybe some other methods

Reply:
ok